dspace-server-webapp/src/main/java/org/dspace/app/rest/security/DSpaceObjectRestPermissionEvaluatorPlugin.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.security;
+
+import java.io.Serializable;
+import java.sql.SQLException;
+import java.util.UUID;
+
+import org.apache.commons.lang3.StringUtils;
+import org.dspace.app.rest.utils.ContextUtil;
+import org.dspace.app.rest.utils.DSpaceObjectUtils;
+import org.dspace.authorize.service.AuthorizeService;
+import org.dspace.content.DSpaceObject;
+import org.dspace.core.Context;
+import org.dspace.services.RequestService;
+import org.dspace.services.model.Request;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Component;
+/**
+ * {@link RestPermissionEvaluatorPlugin} class that evaluate permissions against a generic DSpace Object
+ * 
+ * @author Mykhaylo Boychuk (4science.it)
+ */
+@Component
+public class DSpaceObjectRestPermissionEvaluatorPlugin extends RestObjectPermissionEvaluatorPlugin {
```

Comment:
This PermissionEvaluatorPlugin seems oddly named to me. It looks like it's evaluating for *Admin* permissions on a DSpaceObject. Shouldn't it be named `DSpaceObjectAdminPermissionEvaluatorPlugin` (or something similar, just like the `ResourcePolicyAdminPermissionEvaluatorPlugin`)?